app/src/main/java/io/apicurio/registry/rest/ArtifactsResourceImpl.java

```diff
@@ -451,6 +453,12 @@ public void deleteArtifactRule(RuleType rule, String artifactId) {
     @Override
     public ArtifactMetaData getArtifactMetaData(String artifactId) {
         ArtifactMetaDataDto dto = storage.getArtifactMetaData(artifactId);
+
+        if (dto.getVersion() != ARTIFACT_FIRST_VERSION) {
+            //Fetch the createdOn value from the first version of the artifact
+            ArtifactVersionMetaDataDto artifactVersionMetaData = storage.getArtifactVersionMetaData(artifactId, ARTIFACT_FIRST_VERSION);
+            dto.setCreatedOn(artifactVersionMetaData.getCreatedOn());
+        }
```

Comment:
Doesn't this duplicate the logic in the storage?  It seems like the storage is doing this same work (and I think the storage is where we would want to do this work).

Reply:
Yes, this is part of a previous approach and is something that needs to be removed sorry for that.